src/env.rs

```diff
@@ -12,18 +12,21 @@
 // limitations under the License.
 
 use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::mpsc;
 use std::sync::Arc;
-use std::thread::{self, Builder as ThreadBuilder, JoinHandle};
+use std::thread::{Builder as ThreadBuilder, JoinHandle};
 
 use crate::grpc_sys;
 
-use crate::cq::{CompletionQueue, CompletionQueueHandle, EventType};
+use crate::cq::{CompletionQueue, CompletionQueueHandle, EventType, WorkerInfo};
 use crate::task::CallTag;
 
 // event loop
-fn poll_queue(cq: Arc<CompletionQueueHandle>) {
-    let id = thread::current().id();
-    let cq = CompletionQueue::new(cq, id);
+fn poll_queue(tx: mpsc::Sender<CompletionQueue>) {
+    let cq = Arc::new(CompletionQueueHandle::new());
+    let worker_info = Arc::new(WorkerInfo::new());
+    let cq = CompletionQueue::new(cq, worker_info);
+    let _ = tx.send(cq.clone());
```

Comment:
Because rx is not expected to be dropped before receiving all the completion queues. If error is returned here, then there must be panic inside thread spawn. Dropping result here to avoid double panic. I'm OK to change it to panic here if you feel it's more appropriate.

Reply:
Is it correct that `rx` and `tx` must be on different threads? Then I think double panic is OK.

Reply:
Yes, it is.